nntrainer/engine.h

```diff
@@ -0,0 +1,255 @@
+   *
+   * @param type layer name
+   * @param props property
+   * @return unitque_ptr<T> unique pointer to the Layer object
+   */
+  std::unique_ptr<nntrainer::Layer>
+  createLayerObject(const std::string &type,
+                    const std::vector<std::string> &properties = {}) const {
+    auto ct = getRegisteredContext(parseComputeEngine(properties));
+    ct->getName();
+    return ct->createLayerObject(type);
+  }
+
+  /**
+   * @brief Create an Layer Object with Layer key
+   *
+   * @param int_key key
+   * @param props property
+   * @return unitque_ptr<T> unique pointer to the Layer object
+   */
+  std::unique_ptr<nntrainer::Layer>
+  createLayerObject(const int int_key,
+                    const std::vector<std::string> &properties = {}) const {
+    auto ct = getRegisteredContext(parseComputeEngine(properties));
+    return ct->createLayerObject(int_key);
+  }
+
+  /**
+   * @brief Create an Optimizer Object with Optimizer name
+   *
+   * @param type Optimizer name
+   * @param props property
+   * @return unitque_ptr<T> unique pointer to the Optimizer object
+   */
+  std::unique_ptr<nntrainer::Optimizer>
+  createOptimizerObject(const std::string &type,
+                        const std::vector<std::string> &properties = {}) const {
+    auto ct = getRegisteredContext(parseComputeEngine(properties));
+    return ct->createOptimizerObject(type);
+  }
+
+  /**
+   * @brief Create an Optimizer Object with Optimizer key
+   *
+   * @param int_key key
+   * @param props property
+   * @return unitque_ptr<T> unique pointer to the Optimizer object
+   */
+  std::unique_ptr<nntrainer::Optimizer>
+  createOptimizerObject(const int int_key,
+                        const std::vector<std::string> &properties = {}) const {
+    auto ct = getRegisteredContext(parseComputeEngine(properties));
+    return ct->createOptimizerObject(int_key);
+  }
+
+  /**
+   * @brief Create an LearningRateScheduler Object with type
+   *
+   * @param type type of LearningRateScheduler
+   * @param props property
+   * @return unitque_ptr<T> unique pointer to the LearningRateScheduler object
+   */
+  std::unique_ptr<ml::train::LearningRateScheduler>
+  createLearningRateSchedulerObject(
+    const std::string &type,
+    const std::vector<std::string> &properties = {}) const {
+    auto ct = getRegisteredContext(parseComputeEngine(properties));
+    return ct->createLearningRateSchedulerObject(type, properties);
+  }
+
+  /**
+   * @brief Create an LearningRateScheduler Object with key
+   *
+   * @param int_key key
+   * @param props property
+   * @return unitque_ptr<T> unique pointer to the LearningRateScheduler object
+   */
+  std::unique_ptr<ml::train::LearningRateScheduler>
+  createLearningRateSchedulerObject(
+    const int int_key, const std::vector<std::string> &properties = {}) {
+    auto ct = getRegisteredContext(parseComputeEngine(properties));
+    return ct->createLearningRateSchedulerObject(int_key, properties);
+  }
```

Comment:
it might be wort to swtich this API's to template:
and providing have overloads for:
 `template <typename T> parse(std::vector<std::string> properties)`
 `ct->create<T>(int key, std::vector<std::string>)`
 `ct->create<T>(std::string type, std::string)`

Thus on this level removing all those mostly identical functions, that could be tag dispatched to implementations. Similar idea to how layers are already created in codebase;